model-engine/model_engine_server/api/app.py

```diff
@@ -35,29 +45,31 @@
 app.include_router(file_router_v1)
 app.include_router(trigger_router_v1)
 
-logger = make_logger(filename_wo_ext(__name__))
-
-
-class ExceptionLoggingMiddleware(BaseHTTPMiddleware):
-    async def dispatch(self, request: Request, call_next):
-        try:
-            return await call_next(request)
-        except Exception as e:
-            tb_str = traceback.format_exception(etype=type(e), value=e, tb=e.__traceback__)
-            structured_log = {"error": str(e), "traceback": "".join(tb_str)}
-            logger.error("Unhandled exception: %s", structured_log)
-            request_id = get_request_id()
-            return JSONResponse(
-                {
-                    "status_code": 500,
-                    "content": {
-                        "error": f"Internal error for request_id {request_id}. Our team has been notified."
-                    },
-                }
-            )
 
+@app.middleware("http")
```

Comment:
switched from ASGI middleware to fastapi style middleware in order to try to get DD trace ids to show up (didn't ultimately work, but both middlewares otherwise function the same way)